support-frontend/app/utils/CheckoutValidationRules.scala

```diff
@@ -244,10 +244,19 @@ object AddressAndCurrencyValidationRules {
     if (
       countryFromRequest == Country.US || countryFromRequest == Country.Canada || countryFromRequest == Country.Australia
     ) {
-      stateFromRequest.isDefined.otherwise(s"state is required for $countryFromRequest")
+      {
+        val validState = stateFromRequest match {
+          case None =>
+            Invalid(s"state is required for $countryFromRequest")
+          case Some("") =>
+            Invalid(s"state is required for $countryFromRequest")
+          case _ => Valid
+        }
+        validState
+      }
```

Comment:
I think `match` produces a single expression, so I believe you should be able to remove the val and braces:

```suggestion
      stateFromRequest match {
        case None =>
          Invalid(s"state is required for $countryFromRequest")
        case Some("") =>
          Invalid(s"state is required for $countryFromRequest")
        case _ => Valid
      }
```